src/Consoles/MainConsole.cs

```diff
@@ -1,49 +1,111 @@
+        {
+            base.Draw(delta);
 
-	public class MainConsole : Console
-	{
-	    public Player Player { get; private set; }
-
-		private Room testRoom = new Room (new[]{ new TestEntity () });
-		public MainConsole (int width, int height) : base(width, height)
-		{
-			testRoom.Entities.First ().Renderable = new ConsoleRenderable ('A');
-			testRoom.Entities.First ().Transform.Position = new Point (1,1);
-
-		    Player = new Player("Player");
-		}
-
-		public override void Draw (System.TimeSpan delta)
-		{
-			base.Draw (delta);
-
-			foreach (var entity in testRoom.Entities) {
-				var consoleObject = entity.Renderable.ConsoleObject;
-				consoleObject.Position = entity.Transform.Position;
-				consoleObject.Draw (delta);
-			}
-		}
-
-	    public override bool ProcessKeyboard(Keyboard info)
-	    {
-	        if(info.IsKeyPressed(Keys.Escape))
-	            Screen.MenuConsole.OpenMainMenu();
-
-	        return true;
-	    }
-	}
-}
+            var playerObject = Player.Renderable.ConsoleObject;
+            playerObject.Position = Middle;
+            playerObject.Draw(delta);
+
+            foreach (var entity in testRoom.Entities)
+            {
+                var consoleObject = entity.Renderable.ConsoleObject;
+                consoleObject.Position = entity.Transform.Position - Player.Transform.Position + Middle;
+                if (consoleObject.Position.X < Width && consoleObject.Position.Y < Height)
+                    consoleObject.Draw(delta);
+            }
+        }
+
+        public override void Update(TimeSpan delta)
+        {
+            base.Update(delta);
+            foreach (var entity in testRoom.Entities)
+                (entity as IUpdateable)?.Update(delta);
+            Player.Update(delta);
+        }
+
+        public override bool ProcessKeyboard(Keyboard info)
+        {
+            if (info.IsKeyPressed(Keys.Escape))
+                Screen.MenuConsole.OpenMainMenu();
+
+            if (info.IsKeyDown(Keys.Up))
+            {
+                Player.Transform.Direction = Direction.Up;
+                Player.IsMoving = true;
+            }
+            if (info.IsKeyDown(Keys.Right))
+            {
+                Player.Transform.Direction = Direction.Right;
+                Player.IsMoving = true;
+            }
+            if (info.IsKeyDown(Keys.Left))
+            {
+                Player.Transform.Direction = Direction.Left;
+                Player.IsMoving = true;
+            }
+            if (info.IsKeyDown(Keys.Down))
+            {
+                Player.Transform.Direction = Direction.Down;
+                Player.IsMoving = true;
+            }
+
+            if (info.IsKeyReleased(Keys.Up))
+                Player.IsMoving = false;
+            if (info.IsKeyReleased(Keys.Left))
+                Player.IsMoving = false;
+            if (info.IsKeyReleased(Keys.Right))
+                Player.IsMoving = false;
+            if (info.IsKeyReleased(Keys.Down))
```

Comment:
Chyba powinniśmy sprawdzać jeszcze czy Direction jest odpowiednie, inaczej przy naciskaniue dwóch kierunków nawet jeśli puścimy ten starszy(ten w którym się postać nie porusza) to nadal się ona zatrzyma.